honeybee_vtk/config.py

```diff
@@ -5,200 +5,223 @@
+        description='Boolean value to indicate whether to make the text bold or not.'
     )
 
+    @validator('size')
+    def negative_size_not_allowed(cls, v: int) -> int:
+        if v < 0:
+            raise ValueError('Text size cannot be a negative number.')
+        return v
+
 
 class LegendConfig(BaseModel):
     """Config for the legend to be created from a dataset."""
+    class Config:
+        validate_all = True
+        validate_assignment = True
+
+    color_set: ColorSet = Field(
+        ColorSet.ecotect,
+        description='Color set to be used on data and legend. Currently, this field'
+        ' only uses Ladybug color sets. Defaults to using ecotect colorset.'
+    )
 
-    colors: str = Field(
-        'ecotect',
-        description='Color scheme for the legend.'
+    min: Union[Autocalculate, float] = Field(
+        Autocalculate(),
+        description='Minimum value for the legend. Also known as the lower end of the'
+        ' legend. If min and max values are not specified, autocalculated min and max'
+        ' values will be used from data.'
     )
 
-    range: List[int] = Field(
-        [0, 100],
-        description='A list of minimum and maximum values for the legend.'
+    max: Union[Autocalculate, float] = Field(
+        Autocalculate(),
+        description='Maximum value for the legend. Also known as the higher end of the'
+        ' legend. If min and max values are not specified, autocalculated min and max'
+        ' values will be used from data.'
     )
 
-    show_legend: bool = Field(
+    hide_legend: bool = Field(
         False,
         description='A bool value to indicate whether to show legend in the exported'
         ' images or not.'
     )
 
-    orientation: str = Field(
-        'horizontal',
-        description='Choose between horizontal and vertical orientation.'
-    )
-
-    position: List[float] = Field(
-        [0.5, 0.1],
-        description='A tuple of two decimal values. The values represent the percentage'
-        ' of viewport width and the percentage of viewport height.'
+    orientation: Orientation = Field(
+        Orientation.horizontal,
+        description='Choose between horizontal and vertical orientation of legend.'
     )
 
     width: float = Field(
         0.45,
-        description=' A decimal number representing the percentage of viewport width'
+        description=' A decimal number representing the fraction of viewport width'
         ' that will be used to define the width of the legend.'
     )
 
     height: float = Field(
         0.05,
-        description='A decimal number representing the percentage of viewport height'
+        description='A decimal number representing the fraction of viewport height'
         'that will be used to define the height of the legend.'
     )
 
-    number_of_colors: int = Field(
-        None,
-        description='An integer representing the number of colors in a legend.'
+    position: List[float] = Field(
```

Comment:
I suggest using `confloat` instead of float so that you can enforce that the numbers are between 0 and 1 without the need for another validator.

Reply:
Complied